storage/replica.go

```diff
@@ -1872,6 +1872,16 @@ func (r *Replica) sendRaftMessage(msg raftpb.Message) {
 		return
 	}
 
+	r.store.ctx.Transport.mu.Lock()
+	transportQueues := r.store.ctx.Transport.mu.queues[false]
+	var queuedMsgs int64
+	for _, queue := range transportQueues {
+		queuedMsgs += int64(len(queue))
+	}
+	r.store.metrics.RaftTransportQueueSize.Update(queuedMsgs)
```

Comment:
the metric update should go outside the lock. they have their own synchronization